File: src/main/java/com/sudosoo/takeiteasy/controller/CategoryController.java
package com.sudosoo.takeiteasy.controller;

import com.sudosoo.takeiteasy.dto.category.CategoryResponseDto;
import com.sudosoo.takeiteasy.dto.category.CreateCategoryRequestDto;
import com.sudosoo.takeiteasy.service.CategoryService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/api/category")
@RequiredArgsConstructor
public class CategoryController {
    private final CategoryService categoryService;
    @PostMapping(value = "/create" , name = "createCategory")
    public ResponseEntity<Void> createCategory(@Valid @RequestBody CreateCategoryRequestDto requestDto) {

        categoryService.creatCategory(requestDto);

        return ResponseEntity.ok().build();
    }

    @GetMapping(value = "/getPosts" , name = "getPost")
    public ResponseEntity<CategoryResponseDto> getPosts
            (@RequestParam Long categoryId,
             @RequestParam(required = false, defaultValue = "0", value = "page") int pageNo) {
        pageNo = (pageNo == 0) ? 0 : (pageNo - 1);
        Pageable pageable = PageRequest.of(pageNo,10);
        return new ResponseEntity<>(categoryService.getPostsByCategoryId(categoryId,pageable), HttpStatus.OK);
    }

}

File: takeItEasy-presentation/src/main/kotlin/com/sudosoo/takeItEasy/controller/CategoryController.kt
package com.sudosoo.takeItEasy.controller

import com.sudosoo.takeItEasy.dto.category.CategoryResponseDto
import com.sudosoo.takeItEasy.dto.category.CreateCategoryRequestDto
import com.sudosoo.takeItEasy.service.CategoryService
import com.sudosoo.takeiteasy.dto.category.CategoryResponseDto
import com.sudosoo.takeiteasy.dto.category.CreateCategoryRequestDto
import com.sudosoo.takeiteasy.service.CategoryService
import jakarta.validation.Valid
import org.springframework.data.domain.PageRequest
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.awt.print.Pageable

@RestController
@RequestMapping("/api/category")
class CategoryController (
    val categoryService: CategoryService
) {

    @PostMapping( "/create", name = "createCategory")
    fun createCategory(@Valid @RequestBody requestDto: CreateCategoryRequestDto): ResponseEntity<Void> {
        categoryService.createCategory(requestDto)

        return ResponseEntity.ok().build()
    }

    @GetMapping("/getPosts", name = "getPost")
    fun getPosts(
        @RequestParam categoryId: Long?,
        @RequestParam(required = false, defaultValue = "0", value = "page") pageNo: Int
    ): ResponseEntity<CategoryResponseDto> {
        var pageNo = pageNo
        pageNo = if ((pageNo == 0)) 0 else (pageNo - 1)
        val pageable: Pageable = PageRequest.of(pageNo, 10)
        return ResponseEntity(categoryService.getPostsByCategoryId(categoryId, pageable), HttpStatus.OK)
    }
}